packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -1191,3 +1201,11 @@ class VmServiceCapabilities {
   bool get supportsGetScripts =>
       version.major > 3 || (version.major == 3 && version.minor >= 12);
 }
+
+class ConnectedState {
+  const ConnectedState(this.connected, {this.manual = false});
+
+  final bool connected;
+
+  final bool manual;
```

Comment:
nit: manual isn't the mount obvious name.  Add a comment or maybe change it to 
`intentionalDisconnect` or something ugly but clear.